core/src/main/scala/org/apache/spark/rpc/akka/AkkaRpcEnv.scala

```diff
@@ -214,8 +213,10 @@ private[spark] class AkkaRpcEnv private[akka] (
 
   override def asyncSetupEndpointRefByURI(uri: String): Future[RpcEndpointRef] = {
     import actorSystem.dispatcher
-    actorSystem.actorSelection(uri).resolveOne(defaultLookupTimeout).
-      map(new AkkaRpcEndpointRef(defaultAddress, _, conf))
+    defaultLookupTimeout.addMessageIfTimeout(
+      actorSystem.actorSelection(uri).resolveOne(_).
+        map(new AkkaRpcEndpointRef(defaultAddress, _, conf))
+    )
```

Comment:
I was trying to avoid calling `addMessageIfTimeout` and also still having to reference the `timeout.duration`.  Otherwise, the call would look like this
```scala
defaultLookupTimeout.addMessageIfTimeout(
  actorSystem.actorSelection(uri).resolveOne( defaultLookupTimeout.duration ).
    map(new AkkaRpcEndpointRef(defaultAddress, _, conf))
)
```
but as it is, it does look kind of confusing, so I'm not sure what is better